src/main/java/Task/TaskList.java

```diff
@@ -0,0 +1,248 @@
+    /**
+     * Initializes the tasklist
+     *
+     * @param tasks
+     * @throws DukeException
+     */
+    public TaskList(ArrayList<Task> tasks) throws DukeException {
+
+        this.tasks = tasks;
+        this.addedOrDeletedTask = null;
+    }
+
+    /**
+     * Initializes the tasklist
+     *
+     * @param tasks
+     * @param addedOrDeletedTask
+     */
+    public TaskList(ArrayList<Task> tasks, Task addedOrDeletedTask) {
+        this.tasks = tasks;
+        this.addedOrDeletedTask = addedOrDeletedTask;
+    }
+
+    private static Task assignTask(String type, String name)
+        throws NoTaskException, InvalidCommandException, NoDateException, WrongDateTimeFormatException {
+        try {
+            if (type.equals("todo")) {
+                try {
+                    return new Todo(name.substring(5), false);
+                } catch (IndexOutOfBoundsException e) {
+                    throw new NoTaskException("☹ OOPS!!! The description of a todo cannot be empty.");
+                }
+            } else {
+                int indexOfCommand = name.indexOf("/");
+                String deadline = name.substring(indexOfCommand + 4);
+                if (type.equals("deadline")) {
+                    try {
+                        String currname = name.substring(9);
+                        if (indexOfCommand > -1) {
+                            try {
+                                return new Deadline(name.substring(9, indexOfCommand - 1), false, deadline);
+                            } catch (WrongDateTimeFormatException e) {
+                                throw new WrongDateTimeFormatException(e.getMessage());
+                            }
+                        } else {
+                            throw new NoDateException("☹ OOPS!!! Please specify the deadline!");
+                        }
+                    } catch (IndexOutOfBoundsException e) {
+                        throw new NoTaskException("☹ OOPS!!! The description of a deadline cannot be empty.");
+                    }
+                } else {
+                    try {
+                        String currname = name.substring(6);
+                        if (indexOfCommand > -1) {
+                            return new Event(name.substring(6, indexOfCommand - 1), false, deadline);
+                        } else {
+                            throw new NoDateException("☹ OOPS!!! Please specify when the event is going to be held!");
+                        }
+                    } catch (IndexOutOfBoundsException e) {
+                        throw new NoTaskException("☹ OOPS!!! The description of a event cannot be empty.");
+                    }
+                }
+            }
+        } catch (NullPointerException e) {
+            throw new InvalidCommandException("☹ OOPS!!! I'm sorry, but I don't know what that means :-(");
+        }
+    }
+
+    private static int decideIndexDelete(String word) throws MissingSpecifiedDeleteError {
```

Comment:
Maybe it is better to name the method as decideDeleteIndex as the index is the most concerned word.